src/shogun/mathematics/graph/ops/shogun/details/add_avx.cpp

```diff
@@ -4,43 +4,42 @@
  * Authors: Gil Hoben
  */
 
-#include <unsupported/Eigen/CXX11/Tensor>
+// #include <unsupported/Eigen/CXX11/Tensor>
+
+#include <algorithm>
+#include <functional>
+#include <immintrin.h>
 
 namespace shogun::graph::op {
-	template <typename T>
+	template <typename T1, typename T2=T1>
 	void add_kernel_implementation_avx(
-	    void* input1, void* input2, void* output, const size_t size);
+	    const T2* input1, const T2* input2, T2* output);
 
-	template <typename T>
+	template <typename T1, typename T2=T1>
 	void add_kernel_implementation_avx(
-	    void* input1, void* input2, void* output, const size_t size)
-	{
-		Eigen::TensorMap<Eigen::Tensor<T, 1>> A(static_cast<T*>(input1), size);
-		Eigen::TensorMap<Eigen::Tensor<T, 1>> B(static_cast<T*>(input2), size);
-		Eigen::TensorMap<Eigen::Tensor<T, 1>> Out(static_cast<T*>(output), size);
-
-		Out.device(Eigen::DefaultDevice{}) = A + B;
-	}
-
-	template <>
-	void add_kernel_implementation_avx<bool>(
-	    void* input1, void* input2, void* output, const size_t size)
+	    const T2* input1, const T2* input2, T2* output)
 	{
-		std::transform(
-		    static_cast<const bool*>(input1),
-		    static_cast<const bool*>(input1) + size,
-		    static_cast<const bool*>(input2), static_cast<bool*>(output),
-		    std::plus<bool>());
+		if constexpr (std::is_same_v<T1, float>)
+			*static_cast<__m256*>(output) = _mm256_add_ps(*static_cast<const __m256*>(input1), *static_cast<const __m256*>(input2));
+		else
+		{
+			std::transform(
+			    static_cast<const T2*>(input1),
+			    static_cast<const T2*>(input1) + 32/sizeof(T2),
+			    static_cast<const T2*>(input2), static_cast<T2*>(output),
+			    std::plus<T2>());
```

Comment:
if there is no support for loading/storing the packet carries a void* and the compiler decides how to implement the op